### PR TITLE
ref(hc): organization_integration_added signal in HC

### PR DESCRIPTION
**Pull request by corps**
Enable the `organization_integration_added` signal handling in a cross silo world.

Generalizes signal handling for control -> region signals involving an organization.

### CODE REVIEW

**Comment by corps** [comment]
I changed my mind, I do think I want to generalize this some more.  HRRRRM.

File: src/sentry/api/invite_helper.py [hunk 1]
```diff
@@ -254,6 +255,12 @@ def accept_invite(self, user: User | None = None) -> RpcOrganizationMember | Non
         )
 
         metrics.incr("organization.invite-accepted", sample_rate=1.0)
+        organization_service.schedule_signal(
```
Comment (corps) [inline src/sentry/api/invite_helper.py:258]: After this lands, I can remove the specialized logic in the organization member updated handler https://github.com/getsentry/sentry/blob/master/src/sentry/receivers/outbox/region.py#L110

This is more like how it originally worked.

File: src/sentry/receivers/onboarding.py [hunk 2]
```diff
@@ -539,11 +540,16 @@ def record_issue_tracker_used(plugin, project, user, **kwargs):
...
+        integration_id=integration_id
+    )
+    if integration is None:
+        return
+
     task = OrganizationOnboardingTask.objects.filter(
-        organization_id=organization.id,
+        organization_id=organization_id,
         task=OnboardingTask.INTEGRATIONS,
     ).first()
```
Comment (markstory) [inline src/sentry/receivers/onboarding.py:554]: Should we aim to reduce our usage of signals? They don't lend themselves to readable easy to understand code and I'm concerned about how a generalized approach to sending signals between regions is going to age and perform when we are running multi-region for real.

Should we instead be adapting this kind of logic towards special cased outbox messages? Making operations that involve co-ordination between the region and control silo should be non-trivial to encourage us to find alternate solutions that don't make a tangle of network operations and eventual consistency.

File: src/sentry/models/integrations/integration.py [hunk 3]
```diff
@@ -105,14 +105,22 @@ def add_organization(self, organization: RpcOrganization, user=None, default_aut
...
+                org_integration, created = OrganizationIntegration.objects.get_or_create(
+                    organization_id=organization.id,
+                    integration_id=self.id,
+                    defaults={"default_auth_id": default_auth_id, "config": {}},
+                )
+                # TODO(Steve): add audit log if created
+                if not created and default_auth_id:
+                    org_integration.update(default_auth_id=default_auth_id)
+
+                organization_service.schedule_signal(
```
Comment (scefali) [inline src/sentry/models/integrations/integration.py:118]: @corps hmm shouldn't you send this signal then created is true?

Comment (corps) [reply src/sentry/models/integrations/integration.py:118]: 👍 Can do.

File: src/sentry/receivers/onboarding.py (hunk 2 above)
Comment (corps) [reply src/sentry/receivers/onboarding.py:554]: Talked this over in standup.

File: src/sentry/services/hybrid_cloud/organization/service.py [hunk 4]
```diff
@@ -274,5 +279,40 @@ def update_option(self, *, organization_id: int, key: str, value: OptionValue) -
...
+    def send_signal(
+        self, *, organization_id: int, signal: RpcOrganizationSignal, args: Mapping[str, int | None]
+    ) -> None:
+        pass
+
+    def schedule_signal(
+        self,
+        signal: Signal,
+        organization_id: int,
+        args: Mapping[str, int | None],
```
Comment (RyanSkonnord) [inline src/sentry/services/hybrid_cloud/organization/service.py:293]: Nit: any reason the arg order is different from `send_signal`?

File: src/sentry/services/hybrid_cloud/organization/service.py [hunk 5]
```diff
@@ -274,5 +279,40 @@ def update_option(self, *, organization_id: int, key: str, value: OptionValue) -
...
+    ) -> None:
+        from sentry.models.outbox import ControlOutbox, OutboxCategory, OutboxScope, outbox_context
+
+        if SiloMode.get_current_mode() == SiloMode.REGION:
```
Comment (RyanSkonnord) [inline src/sentry/services/hybrid_cloud/organization/service.py:297]: I think (and you probably agree) that we should avoid looking at the silo mode inside business logic wherever possible, and this is in a gray area as to how much it counts as business logic. I'm fine with this, but if we replicate the pattern into other services ("rule of three" and all), I'd like to try to abstract it out (similar to how `RpcService` abstracts out the mode-sensitive logic for vanilla RPCs).

File: src/sentry/services/hybrid_cloud/organization/service.py (hunk 4 above)
Comment (corps) [reply src/sentry/services/hybrid_cloud/organization/service.py:293]: Addressed

File: src/sentry/services/hybrid_cloud/organization/service.py (hunk 5 above)
Comment (corps) [reply src/sentry/services/hybrid_cloud/organization/service.py:297]: I, agree.  I have an idea, let me get it in.

Comment (corps) [reply src/sentry/services/hybrid_cloud/organization/service.py]: https://github.com/getsentry/sentry/pull/51484/commits/d8f347a1c100fc110970cf68b06322a2d76db4f8

File: src/sentry/models/integrations/integration.py [hunk 6]
```diff
@@ -105,14 +105,23 @@ def add_organization(self, organization: RpcOrganization, user=None, default_aut
...
-            # TODO(Steve): add audit log if created
-            if not created and default_auth_id:
-                org_integration.update(default_auth_id=default_auth_id)
+            with transaction.atomic():
+                org_integration, created = OrganizationIntegration.objects.get_or_create(
+                    organization_id=organization.id,
+                    integration_id=self.id,
+                    defaults={"default_auth_id": default_auth_id, "config": {}},
+                )
+                # TODO(Steve): add audit log if created
```
Comment (scefali) [inline src/sentry/models/integrations/integration.py:114]: lol let me write a card for this

Comment (scefali) [reply src/sentry/models/integrations/integration.py:114]: actually this todo is no longer valid

**Review by markstory — APPROVED** [review]
I don't love signals but we also don't have a better solution at hand and reworking these signals is even more scope to take on.

**Pull request merged**